align/pnr/checker.py

```diff
@@ -141,13 +142,18 @@ def check_place_on_grid(placement_verilog_d, concrete_name, opath):
     else:
         placement_dict = placement_verilog_d
 
+    # If we don't do the copy, then we get the 'flat_leaves" field filled in placement_verilog_d (can be large)
+    # We might want to use an auxillary data structure instead
+
+    placement_verilog_d_copy = copy.deepcopy(placement_verilog_d)
```

Comment:
Deep copy of the emulated data types has indeed been very expensive. Copying just the required subset of data (in this case, only the instances and the transformations) could be more efficient.